modules/core/arrow-extras-data/src/main/kotlin/arrow/data/fingertree/FingerTree.kt

```diff
@@ -0,0 +1,57 @@
+package arrow.data.fingertree
+
+import arrow.core.Option
+import arrow.data.fingertree.internal.FingerTreeInternal
+
+class FingerTree<T> internal constructor(internal: FingerTreeInternal<T> = FingerTreeInternal.Empty()) {
+
+  private var fingerTreeInternal: FingerTreeInternal<T> = internal
+
+  fun append(item: T) {
+    this.fingerTreeInternal = fingerTreeInternal.append(item)
+  }
+
+  fun prepend(item: T) {
```

Comment:
we can't justify having this api that returns Unit like this after 0.9.1. Is there a version of this data structure that has an immutable only API? otherwise this may not belong in Arrow as it deviates from the rest of the library which exclusively offers pure APIs except for unsafe runners in the effects library. Not saying this is isn't useful just not in `arrow.data`.

Reply:
FTR internal mutability for optimizations is fine but not public functions that return `Unit` and mutate the internal state of data they are part of.

Reply:
`FingerTreeInternal` is the immutable only API. I've created the `FingerTree` wrapper class for convenient use of the internal finger tree, but I totally get you point!
So I would say that we just get rid of the wrapper and make the internal finger tree non-internal?

Reply:
@juliankotrba yes, I believe that would be a much better fit in Arrow. It will help to see at a glance a list of functions in the public API to understand where this will best fit and if it can provide extensions for some of type classes as well.